contracts/StablecoinConverter.sol

```diff
@@ -183,7 +186,19 @@ contract StablecoinConverter is EpochTokenLocker {
         uint128 buyAmount,
         uint128 sellAmount
     ) public returns (uint256) {
-        return placeValidFromOrder(buyToken, sellToken, getCurrentBatchId(), validUntil, buyAmount, sellAmount);
+        uint32 validFrom = getCurrentBatchId();
```

Comment:
This code duplication! nooooo.

But all other variants that I can think of are more gas expensive.

Hm... let me think a little bit more about it

Reply:
Is this comment about this particular line of code, or just a comment at an arbitrary point in the code?

Reply:
Can we have this instead:
```
    function placeOrderInternal(
        uint16 buyToken,
        uint16 sellToken,
        uint32 validUntil,
        uint128 buyAmount,
        uint128 sellAmount
    ) internal returns (uint256) {
        // ...
    }

    function placeOrder(
        uint16 buyToken,
        uint16 sellToken,
        uint32 validUntil,
        uint128 buyAmount,
        uint128 sellAmount
    ) internal returns (uint256) {
         return placeOrderInternal(
                buyToken,
                sellToken,
                validUntil,
                buyAmount,
                sellAmount
            );
    }
    
    function placeValidFromOrder(
        uint16[] memory buyTokens,
        uint16[] memory sellTokens,
        uint32[] memory validUntils,
        uint128[] memory buyAmounts,
        uint128[] memory sellAmounts
    ) public returns (uint256[] memory orderIds) {
        uint256 length = buyTokens.length;
        require(sellTokens.length == length);
        require(validUntils.length == length);
        require(buyAmounts.length == length);
        require(sellAmounts.length == length);
        
        orderIds = new uint256[](length);
        for (uint256 i = 0; i < length; i++) {
            orderIds[i] = placeOrderInternal(
                buyTokens[i],
                sellTokens[i],
                validUntils[i],
                buyAmounts[i],
                sellAmounts[i]
            );
        }
    }
```

Reply:
From what I understand, internal functions are just EVM jumps and are gas efficient.

Reply:
Ya Great! I will go with that! Thanks a millio!

Reply:
I am not going to include all those require statements, because it will drastically increase our tests and isn't super necessary (from my perspective). Does anyone have a strong opinion about that?

Also. What should I return from the multi-order placement?

Reply:
Aren't the require statements required? What happens when you try to index an array out of bounds?

Reply:
Ya... I guess.